core/controllers/profile.py

```diff
@@ -454,7 +454,7 @@ def get(self):
         else:
             if self.request and self.request.get('current_url'):
                 target_url = (
-                    '/' if self.request.get('current_url').endswith(
+                    '/splash' if self.request.get('current_url').endswith(
```

Comment:
Why this change? This is a sensitive part of the code, please analyze in detail and explain carefully the rationale for what changes should be made here.

Reply:
Actually, this was kinda confusing. Before these URL changes, the routing worked like this:
`/splash` requests would always lead to the splash page(home page). But requests to `/` would use the HomePageRedirectPage handler and would check if the user is logged in or not. If they are, they would be led to their default dashboard(creator/learner) and if they aren't, they would be led to the splash page(home page).

Now, since we want the home page to be redirected by `/` requests, we need something else for the automatic dashboard redirection. So, I've used the old `/splash` URL for that. It redirects the user to their default dashboard if they're logged in. Simply put, the roles are reversed now. The `/splash` URL leads to `/` if not logged in else to dashboard and the `/` URL always leads to the home page.

The advantage? Now, the `/splash` URL is more or less hidden and is only used in the backed for appropriate redirection when the user logs in or clicks on the logo on the top left. `/` will be the primary URL now.

Please let me know if I've missed something.

Reply:
Thanks for the clear explanation. I think this is something that you should discuss with Danial and Diana -- about the difference in behaviour for logged in and logged out users and what the correct approach should be. 

In particular, is it OK if the splash page is not visible at all to logged-in users? If not, we can always redirect / to the appropriate page for logged in users.

Reply:
Diana probably has a stronger opinion. What I think is your recommendation is alright, meaning the logged in user stays on the dashboard when clicking on the LOGO. But, a button like "library" as already exists is necessary to let the user explore other lessons. Probably having an "add new lesson" button in the dashboard would help.

Reply:
@dchen97 thoughts?

Also, @Danialp87, about the "add new lesson" button, isn't that purpose being served by the "CREATE" button that currently exists and allows users to create new explorations/collections?

Reply:
No indeed, I meant new lesson button for learners.

Reply:
I think this behavior is fine for now as the signed in users are likely already familiar with Oppia to a certain extent and should just be allowed to start taking action (i.e. creating and/or taking lessons). However, I'm a bit hesitant about not letting logged in users access the home page _at all_ so perhaps we can follow this up with an improved home page experience that incorporates elements / links from the home page but also presents dashboard elements if the user is logged in.

Reply:
Just an update -- I talked with @dchen97 about this and our conclusion was that we just need to ensure that everything that's accessible from the "logged-out homepage" is accessible to logged-in users via other means (like the About menu). This should be the case, so I don't think that further action is needed.

Reply:
Thanks, guys! Just confirming, the scenario should be:
User goes to oppia.org(regardless of login status)-- > The splash page opens up
User logs in --> Default dashboard opens up
User clicks on the logo-- > Default dashboard(if logged in, else splash page) opens up
User logs out-- > The splash page opens up

And, in none of these scenarios will 'splash' be in the URL.

/ cc @seanlip @Danialp87 @dchen97

Reply:
"User goes to oppia.org(regardless of login status)" -- if the user is logged in they see the splash page. If the user is not logged in they see their default dashboard. The behaviour should be the same as if they click on the logo.

For "User logs out" -- I think the "splash page opens up" is fine. But, alternatively, if the user is taken back to the page they were on just before, that's fine too. Maybe just do whatever's in the codebase at the moment / requires minimal changes.

Everything else is correct. Thanks!

Reply:
> "User goes to oppia.org(regardless of login status)" -- if the user is logged in they see the splash page. If the user is not logged in they see their default dashboard. 

Um...you mean the opposite, right? The user would see the default dashboard only when they are logged in, or am I missing something?
Also, this is how it does and will work for the logo, but don't you think the user should be taken to the splash page regardless of whether they are logged in or not? Since otherwise there would be no way for a logged-in user to access the splash page.

Reply:
Sorry, yes, I meant the opposite, thanks for catching that. And I've discussed with @dchen97 and @Danialp87 and it's fine for there to be no way for a logged-in user to access the splash page.